packages/dev-tools/src/actions/WriteSummary.tsx

```diff
@@ -62,9 +68,19 @@ export function WriteSummary({ write }: Props) {
     const functionSelectorAndArgs: Hex = write.request?.args?.length
       ? (write.request.args[write.request.args.length - 1] as Hex)
       : `0x`;
-    const functionData = decodeFunctionData({ abi: worldAbi, data: functionSelectorAndArgs });
-    functionName = functionData.functionName;
-    functionArgs = functionData.args;
+
+    // TODO: Since `functionSelectorAndArgs` corresponds to a System's function, decoding it using
+    // the World ABI may not always be successful. For instance, namespaced system calls could
+    // result in an error.
+    try {
+      const functionData = decodeFunctionData({ abi: worldAbi, data: functionSelectorAndArgs });
+      functionName = functionData.functionName;
+      functionArgs = functionData.args;
+    } catch (error) {
+      if (!(error instanceof AbiFunctionSignatureNotFoundError)) {
+        throw error;
+      }
+    }
```

Comment:
I've found an error in dev-tools. I've made a workaround, but I'm not sure what's the best way to address this, so leaving this as TODO now.

For root namespaced calls:
<img width="613" alt="Screenshot 2024-03-05 at 21 39 55" src="https://github.com/latticexyz/mud/assets/81064017/9deca912-efef-4c6a-ad4c-8c583e7fee8d">

For non-root namespaced calls (This was an error before this change):
<img width="612" alt="Screenshot 2024-03-05 at 21 35 54" src="https://github.com/latticexyz/mud/assets/81064017/adf5ce15-b7bd-4e62-9f65-28cbab4560a3">

Reply:
we could try to do the inverse, where we resolve the system function selector form the world one and assume the same args?

Reply:
> we don't have the system signature

right 🤔

Reply:
added an issue to track this here: https://github.com/latticexyz/mud/issues/2382